src/target/stm32f4.c

```diff
@@ -207,36 +207,37 @@ static void stm32f4_detach(target *t)
 
 bool stm32f4_probe(target *t)
 {
-	if (t->part_id == ID_STM32F20X) {
-		/* F405 revision A have a wrong IDCODE, use ARM_CPUID to make the
-		 * distinction with F205. Revision is also wrong (0x2000 instead
-		 * of 0x1000). See F40x/F41x errata. */
-		if ((t->cpuid & 0xFFF0) == CORTEX_M4)
-			t->part_id = ID_STM32F40X;
-	}
-	switch(t->part_id) {
+	const uint16_t mcu_idcode = target_mem_read32(t, DBGMCU_IDCODE) & 0xfffU;
```

Comment:
Because of the assignment in the first case, this can't be marked const (unfortunatly)

Reply:
I see. But there is an error because in switch-case we have 2 identical cases. So I returned if statement before switch-case

Reply:
Ah, good catch. We'll have a play later with seeing if we can still somehow remove the need for the pre-switch if (think we know a way), but getting this fix in is more important right now.